components/org.wso2.identity.event.http.publisher/src/main/java/org/wso2/identity/event/http/publisher/internal/component/HTTPAdapterServiceComponent.java

```diff
@@ -0,0 +1,120 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.identity.event.http.publisher.internal.component;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.wso2.carbon.identity.event.publisher.api.service.EventPublisher;
+import org.wso2.carbon.identity.webhook.management.api.service.WebhookManagementService;
+import org.wso2.carbon.identity.webhook.metadata.api.exception.WebhookMetadataException;
+import org.wso2.carbon.identity.webhook.metadata.api.service.EventAdapterMetadataService;
+import org.wso2.identity.event.http.publisher.internal.config.HTTPAdapterConfiguration;
+import org.wso2.identity.event.http.publisher.internal.service.impl.HTTPEventPublisherImpl;
+
+import static org.wso2.carbon.identity.webhook.metadata.internal.constant.ErrorMessage.ERROR_CODE_ADAPTER_NOT_FOUND;
+import static org.wso2.identity.event.http.publisher.internal.constant.HTTPAdapterConstants.HTTP_ADAPTER_NAME;
+
+/**
+ * HTTP Outbound Event Adapter service component.
+ */
+@Component(
+        name = "org.wso2.identity.event.http.publisher.internal.component.HTTPAdapterServiceComponent",
+        immediate = true)
+public class HTTPAdapterServiceComponent {
+
+    private static final Log log = LogFactory.getLog(HTTPAdapterServiceComponent.class);
+
+    @Activate
+    protected void activate(ComponentContext context) {
+
+        try {
+            HTTPAdapterDataHolder.getInstance().setAdapterConfiguration(new HTTPAdapterConfiguration(
+                    HTTPAdapterDataHolder.getInstance().getEventAdapterMetadataService()
+                            .getAdapterByName(HTTP_ADAPTER_NAME).getProperties()));
+            if (HTTPAdapterDataHolder.getInstance().getAdapterConfiguration().isAdapterEnabled()) {
+                // Register EventPublisher service
+                HTTPEventPublisherImpl eventPublisherService = new HTTPEventPublisherImpl();
+                context.getBundleContext().registerService(EventPublisher.class.getName(),
+                        eventPublisherService, null);
+                HTTPAdapterDataHolder.getInstance().setClientManager(new ClientManager());
+                log.debug("Successfully activated the HTTP adapter service.");
+            } else {
+                log.error("HTTP Adapter is not enabled.");
```

Comment:
This is not an error case. It's a debug scenario or a warn scenario.
Let's keep it as a debug

Reply:
Also does this else block is reached ? why we have an exception handled for the same case below if so